src/dom.js

```diff
@@ -204,6 +204,49 @@ export function closestByTag(element, tagName) {
   });
 }
 
+/**
+ * Finds the closest element with the specified selector from this element
+ * @param {!Element} element
+ * @param {string} selector
+ * @return {?Element} closest ancestor if found.
+ */
+export function closestBySelector(element, selector) {
+  if (element.closest) {
+    return element.closest(selector);
+  }
+
+  return closest(element, el => {
+    return matches(el, selector);
```

Comment:
I'm not sure we're talking about the same thing? I mean do we need to code a fallback cb if `matches` is not supported natively.